app/controllers/hackdays_controller.rb

```diff
@@ -0,0 +1,77 @@
+class HackdaysController < ApplicationController
+
+  def index
+    @hackday = Hackday.new
+    set_current_and_past_hackdays
+  end
+
+  def create
+    @hackday = Hackday.new hackday_params
+    
+    if @hackday.save
+      cookies["#{@hackday.id}_votes"] = MAX_VOTES
+      redirect_to @hackday
+    else
+      set_current_and_past_hackdays
+      render :index
+    end
+  end
+
+  def edit
+    @hackday = Hackday.find(params[:id])
+    unless @hackday.active?
+      flash[:warning] = "Cannot edit a non-active hackday"
+      redirect_to request.referrer || root_url
```

Comment:
if you are going to do || you should really use brackets to make it clearer. so it should be:

redirect_to(request.referrer || root_url)